lib/rex/proto/kerberos/crypto.rb

```diff
@@ -4,15 +4,80 @@ module Rex
   module Proto
     module Kerberos
       module Crypto
+        module KeyUsage
+          AS_REQ_PA_ENC_TIMESTAMP                        = 1
+          KDC_REP_TICKET                                 = 2
+          AS_REP_ENCPART                                 = 3
+          TGS_REQ_KDC_REQ_BODY_AUTHDATA_SESSION_KEY      = 4
+          TGS_REQ_KDC_REQ_BODY_AUTHDATA_SUB_KEY          = 5
+          TGS_REQ_PA_TGS_REQ_AP_REQ_AUTHENTICATOR_CHKSUM = 6
+          TGS_REQ_PA_TGS_REQ_AP_REQ_AUTHENTICATOR        = 7
+          TGS_REP_ENCPART_SESSION_KEY                    = 8
```

Comment:
Just to clarify, this comes from the key usage numbers here right? 👀 
https://datatracker.ietf.org/doc/html/rfc4120#section-7.5.1

As I understand it, the TGS_REP_ENC_PART contains session_key, but also other fields like srealm etc.

```
   EncTGSRepPart   ::= [APPLICATION 26] EncKDCRepPart

   EncKDCRepPart   ::= SEQUENCE {
           key             [0] EncryptionKey,
           last-req        [1] LastReq,
           nonce           [2] UInt32,
           key-expiration  [3] KerberosTime OPTIONAL,
           flags           [4] TicketFlags,
           authtime        [5] KerberosTime,
           starttime       [6] KerberosTime OPTIONAL,
           endtime         [7] KerberosTime,
           renew-till      [8] KerberosTime OPTIONAL,
           srealm          [9] Realm,
           sname           [10] PrincipalName,
           caddr           [11] HostAddresses OPTIONAL
   }
```

So I believe this would just be:
```suggestion
          TGS_REP_ENCPART            = 8
```

Reply:
Ah, I see. The suffix is to denote which key/subkey is being used to encrypt the ciphertext